pubsub/runtime/src/main/java/io/quarkiverse/googlecloudservices/pubsub/push/GoogleTokenVerifier.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkiverse.googlecloudservices.pubsub.push;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Optional;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+
+/**
+ * Default implementation of {@link TokenVerifier} using the {@link GoogleIdTokenVerifier}.
+ */
+@ApplicationScoped
+public class GoogleTokenVerifier implements TokenVerifier {
```

Comment:
Is it really needed to create our own implementation?
This seems to only delegate to the GoogleIdTokenVerifier so a simple producer of this type could be used instead.